hnn_core/gui/_viz_manager.py

```diff
@@ -762,13 +760,10 @@ def _add_figure(b, widgets, data, template_type, scale=0.95, dpi=96):
                    scale * ((int(viz_output_layout.height[:-2]) - 10) / dpi))
         mosaic = template_type['mosaic']
         kwargs = template_type['kwargs']
-        plt.ioff()
-        fig, axd = plt.subplot_mosaic(mosaic,
-                                      figsize=figsize,
-                                      dpi=dpi,
-                                      **kwargs)
-        plt.ion()
-        fig.tight_layout()
+        with plt.ioff():
```

Comment:
what's the significance of this context manager?

Reply:
It turns off interactive mode. With interactive on the plot may render each time a change is made. Switching it off will construct the figure object but won't display/render it until a plt.show() is called. 

That said... the fig.canvas commands after this context should be within the context manager here....